float32 temperature
float32 stir_speed
float32 viscosity_trend
